### PR TITLE
tx redirect & tx feedback

### PR DESCRIPTION
**Pull request by pldespaigne**

### CODE REVIEW

File: libs/ethers/src/lib/wallet/wallet-add-key/wallet-add-key.component.ts [hunk 1]
```diff
@@ -90,7 +83,7 @@ export class WalletAddKeyTunnelComponent implements OnInit {
     if (!!this.walletQuery.getValue().tx) {
       this.router.navigateByUrl('/layout/o/account/wallet/send');
     } else {
-      this.router.navigateByUrl(this.redirectRoute);
+      this.router.navigateByUrl('/layout/o/account/wallet');
```
Comment (Maxstgt) [inline libs/ethers/src/lib/wallet/wallet-add-key/wallet-add-key.component.ts]: I can't see the scope, but why don't use the `walelt.feedback.redirectRoute` for it? I'm not sure if it is possible without an extra import. But you are using it in the HTML

File: libs/organization/src/lib/components/member-form-role/member-form-role.component.ts [hunk 2]
```diff
@@ -63,13 +63,29 @@ export class MemberFormRoleComponent {
       this.permissionsService.updateMembersRole(members);
     };
 
+    const orgName = this.query.getValue().org.name;
+    const orgId = this.query.getValue().org.id;
```
Comment (Maxstgt) [inline libs/organization/src/lib/components/member-form-role/member-form-role.component.ts]: Just a suggestion but there is getter for it. `this.query.id`

File: libs/organization/src/lib/components/organization-form-operation/organization-form-operation.component.ts [hunk 3]
```diff
@@ -39,14 +40,24 @@ export class OrganizationFormOperationComponent {
...
+    const removedMember = this.control.get('members').value.find(member => member.uid === id) as OrganizationMember;
     const members = this.control.get('members').value.filter(member => member.uid !== id);
     this.control.get('members').patchValue(members);
 
     const memberAddress = await this.service.getMemberAddress(removedMember.email);
     const orgAddress = await this.service.getAddress();
     const operationId = this.control.get('id').value;
-    this.walletService.setTx(CreateTx.removeMember(orgAddress, operationId, memberAddress));
+    const operationName = this.control.get('name').value;
+    const orgId = this.query.getValue().org.id;
```
Comment (Maxstgt) [inline libs/organization/src/lib/components/organization-form-operation/organization-form-operation.component.ts]: Same suggestion as above

File: libs/organization/src/lib/components/organization-form-operation/organization-form-operation.component.ts [hunk 4]
```diff
@@ -59,15 +70,35 @@ export class OrganizationFormOperationComponent {
     const memberAddress = await this.service.getMemberAddress(addedMember.email);
     const orgAddress = await this.service.getAddress();
     const operationId = this.control.get('id').value;
-    this.walletService.setTx(CreateTx.addMember(orgAddress, operationId, memberAddress));
+    const operationName = this.control.get('name').value;
+    const orgId = this.query.getValue().org.id;
```
Comment (Maxstgt) [inline libs/organization/src/lib/components/organization-form-operation/organization-form-operation.component.ts]: ;)

File: libs/organization/src/lib/components/organization-form-operation/organization-form-operation.component.ts [hunk 5]
```diff
@@ -59,15 +70,35 @@ export class OrganizationFormOperationComponent {
...
     this.router.navigateByUrl('/layout/o/account/wallet/send');
   }
 
   public async updateQuorum() {
     const orgAddress = await this.service.getAddress();
     const operationId = this.control.get('id').value;
     const newQuorum = this.control.get('quorum').value;
-    this.walletService.setTx(CreateTx.modifyQuorum(orgAddress, operationId, newQuorum));
+    const operationName = this.control.get('name').value;
+    const orgId = this.query.getValue().org.id;
```
Comment (Maxstgt) [inline libs/organization/src/lib/components/organization-form-operation/organization-form-operation.component.ts]: :P

File: libs/organization/src/lib/components/organization-signer-form/organization-signer-form.component.ts [hunk 6]
```diff
@@ -49,11 +50,32 @@ export class OrganizationSignerFormComponent {
 
     const memberAddress = await this.service.getMemberAddress(this.member.email);
     const orgAddress = await this.service.getAddress();
+    const memberName = this.member.name;
+    const operationName = currentOperation.name;
+    const orgId = this.query.getValue().org.id;
```
Comment (Maxstgt) [inline libs/organization/src/lib/components/organization-signer-form/organization-signer-form.component.ts]: :)

File: libs/ethers/src/lib/wallet/wallet-add-key/wallet-add-key.component.ts (hunk 1 above)
Comment (pldespaigne) [reply libs/ethers/src/lib/wallet/wallet-add-key/wallet-add-key.component.ts]: I use `wallet.feedback.redirectRoute` to redirect the user at the end of a *send tx tunnel*, here we are in the *add key tunnel*, in this tunnel I want to either go back to the wallet (if the user came from here) or go back to sending a tx (if the user was doing that, if it's the case, then the tx tunnel will redirect correctly at the end). I hope it's clear ^^

**Pull request merged**